lib/src/widget/nav_bar/widget/nav_bar.dart

```diff
@@ -39,48 +39,93 @@ class NavBarState extends State<NavBar> implements ConfigUpdater {
     );
 
     setState(() {
+      desiredWidth = 0;
+      iconSize = linden.dimen.navBarItemHeight;
       this.config = config;
     });
   }
 
-  @override
-  Widget build(BuildContext context) {
-    final config = this.config;
-    if (config == null || config.type.isHidden) return const Center();
+  double desiredWidth = 0;
+  double maxPossibleWidth = 0;
 
-    final items = config.items.map(_buildItem).toList(growable: false);
+  double get itemSidePadding => linden.dimen.unit0_25;
 
-    if (config.type == NavBarType.backBtn) {
-      return _withPadding(_withFixedHeight(Row(children: [items.first])));
-    }
+  late double iconSize = linden.dimen.navBarItemHeight;
 
-    final row = Row(
-      mainAxisSize:
-          config.isWidthExpanded ? MainAxisSize.max : MainAxisSize.min,
-      mainAxisAlignment: config.isWidthExpanded
-          ? MainAxisAlignment.spaceBetween
-          : MainAxisAlignment.start,
-      children: items,
-    );
+  double get navBarSidePadding => linden.dimen.unit;
 
-    // AppCardWidget can't be used becasue it uses Material widget
-    // with MaterialType.card, which doesn't allow to customise shadows.
-    final card = Container(
-      padding: EdgeInsets.symmetric(horizontal: linden.dimen.unit),
-      child: row,
-      decoration: BoxDecoration(
-        color: linden.colors.background,
-        borderRadius: linden.styles.roundBorderCard,
-        boxShadow: [linden.styles.cardShadow],
-      ),
-    );
+  bool useDefaultIconSize = true;
+
+  bool get shouldIconBeSmaller {
+    final shouldBeSmaller = desiredWidth > maxPossibleWidth;
+    if (useDefaultIconSize && shouldBeSmaller) useDefaultIconSize = false;
+    return shouldBeSmaller;
+  }
 
-    final sized = _withFixedHeight(card);
+  void _checkDesiredWidth(int itemsCount) {
+    desiredWidth =
+        // number of items * by its size
+        (iconSize * itemsCount) +
+            // each item has 2 paddings (left and right) except first and last items
+            // those have only 1, so we need to remove 2 paddings
+            ((itemSidePadding * itemsCount) -
+                itemSidePadding -
+                itemSidePadding) +
+            itemsCount * itemSidePadding +
+
+            // side padding inside the bar
+            navBarSidePadding * 2 +
+
+            // padding from left and right
+            widget.padding.left +
+            widget.padding.right;
+    while (shouldIconBeSmaller) {
+      iconSize -= 1;
+      _checkDesiredWidth(itemsCount);
+    }
+  }
```

Comment:
🛑 I am not sure you need to use a while loop:

```
maxPossibleWidth = (iconSize * itemCount) + paddings;
maxPossibleWidth - paddings = (iconSize * itemCount) 
(maxPossibleWidth - paddings) / itemCount = iconSize 

```

so 
`iconSize  = ((maxPossibleWidth - paddings) / itemCount).floor()`

Is a bit more cleaner and it avoids the dangerous while loop.

Reply:
updated this code